cli/utils.py

```diff
@@ -49,9 +46,9 @@ def none_resolver(first, second, key):
 
         # tyr to merge lists first
         if isinstance(first[key], list):
-            if isinstance(first[key], list):
+            if isinstance(second[key], list):
                 first[key].extend(second[key])
-            else:
+            elif second[key] is not None:
                 first[key].append(first[key])
 
         if key not in first or first[key] is None:
```

Comment:
Comments about dict_merge:

- dict merge strategy is unclear. need more doc. testing would be helpful as well.
- let's avoid merging lists. values should be overwritten
- the recursive call to dict_merge doesn't use the conflict resovler
- Unless we have a need for different conflict resolvers, I believe that we can allways use the default one.
- If I understand correctly, first[key] is chosen unless it is `None`, in which case, second[key] is chosen. I suggest to always use use second[key] unless:
  - key not in second
  - both are dicts and then invoke recursion.